node_modules/oae-core/signin/js/signin.js

```diff
@@ -0,0 +1,196 @@
+         */
+        var validateLocalSignInForm = function() {
+            var validateOpts = {
+                'messages': {
+                    'signin-local-username': {
+                        'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_YOUR_USERNAME__'),
+                        'signin-local-failed-attempt': oae.api.i18n.translate('__MSG__INVALID_USERNAME_OR_PASSWORD__')
+                    },
+                    'signin-local-password': oae.api.i18n.translate('__MSG__PLEASE_ENTER_YOUR_PASSWORD__')
+                },
+                'methods': {
+                    'signin-local-failed-attempt': {
+                        'method': function(value, element) {
+                            // This class will be added after a failed login attempt and is used
+                            // to tell jquery.validate to mark the field as invalid
+                            return false;
+                        }
+                    }
+                },
+                'submitHandler': doLocalSignIn
+            };
+
+            oae.api.util.validation().validate($('#signin-local-form', $rootel), validateOpts);
+        };
+
+        /**
+         * Render the sign in options (external auth options, username & password form, etc...)
+         * form
+         */
+        var renderSignInOptions = function() {
+            oae.api.util.template().render($('#signin-options-template', $rootel), {
+                'authStrategyInfo': authStrategyInfo,
+                'redirectUrl': signInRedirectUrl,
+                'invitationInfo': invitationInfo,
+                'showSignUp': showSignUp
+            }, $('#signin-options-container', $rootel));
+
+            // Handle username / password authentication
+            validateLocalSignInForm();
+        };
+
+        /**
+         * Render the footer for the signin modal
+         */
+        var renderSignInFooter = function() {
+            oae.api.util.template().render($('#signin-modal-footer-template', $rootel), {
+                'showSignUp': showSignUp
+            }, $('#signin-modal-footer-container', $rootel));
+        };
+
+        /**
+         * Initialize the signin modal
+         */
+        var init = function() {
+            $(document).on('click', '.oae-trigger-signin', function() {
+                authStrategyInfo = oae.api.authentication.getStrategyInfo();
+
+                // Get the sign in options from the trigger element
+                var signInData = $(this).data();
+                signInRedirectUrl = signInData.redirecturl;
+                invitationInfo = {
+                    'email': signInData.email,
+                    'token': signInData.token
+                };
+
+                if (signInData.hidesignup) {
+                    showSignUp = false;
+                }
+
+                // Auto-signin if there is only one external signin method available
+                if (authStrategyInfo.hasSingleExternalAuth) {
```

Comment:
Shouldn't this only execute when there is a single external authentication strategy and no local auth ?